src/pepr/operator/controllers/keycloak/authservice/config.ts

```diff
@@ -156,3 +252,43 @@ async function checksumDeployment(checksum: string) {
     throw new Error("Failed to apply the checksum to authservice", { cause: e });
   }
 }
+
+/**
+ * Applies a checksum to the Kubernetes authservice deployment to force a rollout.
+ *
+ * @param {string} base64EncodedConfig - The base64 encoded AuthserviceConfig to apply.
+ */
+async function applySecret(base64EncodedConfig: string) {
+  try {
+    return await K8s(kind.Secret)
+      .Apply(
+        {
+          metadata: {
+            namespace: operatorConfig.namespace,
+            name: operatorConfig.secretName,
+          },
+          data: {
+            "config.json": base64EncodedConfig,
+          },
+        },
+        { force: true },
+      )
+      .then(secret => JSON.parse(atob(secret.data!["config.json"])) as AuthserviceConfig);
+  } catch (e) {
+    log.error(`Failed to apply the authservice config secret: ${e.data?.message}`);
```

Comment:
No guarantee that the error has the `data` structure, should we have default message here for better error handling? This might be applicable in a couple other places as well.

Reply:
Updated